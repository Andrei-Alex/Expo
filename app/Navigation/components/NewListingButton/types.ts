export interface INewListingButton {
  onPress: () => void;
  testID?: string;
}
