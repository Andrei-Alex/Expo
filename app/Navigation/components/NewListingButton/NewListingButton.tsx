import * as React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { styles, INewListingButton } from '.';
import { colors } from '../../../config';

/**
 * NewListingButton Component
 *
 * A customizable button component for React Native navigation, specifically designed for actions related to new listings.
 * It prominently features a plus-circle icon from the MaterialCommunityIcons set, signaling an action to add or create new items.
 * The component is wrapped in a `TouchableOpacity` for enhanced user interaction, allowing for feedback on touch with the
 * onPress prop. It is styled with a predefined `styles.container` and utilizes the `colors.white` for the icon color, ensuring
 * the button is both visually appealing and clearly visible.
 *
 * ### Usage
 *
 * ```jsx
 * <NewListingButton onPress={() => console.log('Button pressed!')} />
 * ```
 *
 * @component
 * @param {INewListingButton} props - The props passed to the component.
 * @param {Function} props.onPress - A callback function triggered when the button is pressed. This function should handle the desired action for creating a new listing.
 * @param {string} [props.testID='newListingButton'] - An optional ID used for testing purposes. Default is 'newListingButton'.
 * @returns {React.ReactElement} A stylized, interactive button component featuring a plus-circle icon, intended for adding new listings or similar functions in a React Native app.
 *
 **/

const NewListingButton: React.FC<INewListingButton> = ({
  onPress,
  testID = 'newListingButton'
}) => {
  return (
    <TouchableOpacity testID={testID} onPress={onPress}>
      <View style={styles.container}>
        <MaterialCommunityIcons
          name={'plus-circle'}
          color={colors.white}
          size={30}
        />
      </View>
    </TouchableOpacity>
  );
};

export default NewListingButton;
