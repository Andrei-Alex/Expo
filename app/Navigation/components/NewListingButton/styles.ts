import { StyleSheet } from 'react-native';
import { colors } from '../../../config';

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.primary,
    borderColor: colors.white,
    borderWidth: 10,
    borderRadius: 40,
    height: 70,
    width: 70,
    bottom: 20
  }
});
