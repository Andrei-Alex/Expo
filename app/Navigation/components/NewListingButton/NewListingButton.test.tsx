import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import NewListingButton from './NewListingButton';
import * as renderer from 'react-test-renderer';

describe('NewListingButton', () => {
  const onPressMock = jest.fn();

  it('render correctly', () => {
    const component = renderer.create(
      <NewListingButton onPress={onPressMock} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('call on press function', () => {
    const { getByTestId } = render(<NewListingButton onPress={onPressMock} />);
    const newButton = getByTestId('newListingButton');

    fireEvent.press(newButton);
    expect(onPressMock).toHaveBeenCalled();
  });
});
