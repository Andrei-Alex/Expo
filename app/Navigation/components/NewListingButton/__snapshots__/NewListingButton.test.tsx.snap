// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListingButton render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="newListingButton"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fc5c65",
        "borderColor": "#fff",
        "borderRadius": 40,
        "borderWidth": 10,
        "bottom": 20,
        "height": 70,
        "justifyContent": "center",
        "width": 70,
      }
    }
  >
    <Text />
  </View>
</View>
`;
